v0/destinations/branch/transform.js

```diff
@@ -81,72 +59,98 @@ function getCategoryAndName(rudderEventName) {
 }
 
 function getUserData(message) {
-  const { context } = message;
-  const { os } = message.context;
+  let userData = {
+    os: get(message, "context.os.name"),
+    os_version: get(message, "context.os.version"),
+    app_version: get(message, "context.app.version"),
+    model: get(message, "context.device.model"),
+    brand: get(message, "context.device.brand"),
+    screen_dpi: get(message, "context.screen.density"),
+    screen_height: get(message, "context.screen.height"),
+    screen_width: get(message, "context.screen.width"),
+    developer_identity: getFieldValueFromMessage(message, "userId"),
+    user_agent: get(message, "context.userAgent")
+  };
 
-  if (!os || !isDefinedAndNotNullAndNotEmpty(os.name)) {
-    throw new CustomError(
-      "os name is missing in the payload and please make sure to insert it at context.os.name",
-      400
-    );
+  const os = get(message, "context.os.name");
+
+  if (isDefinedAndNotNull(os)) {
+    if (isAppleFamily(os)) {
+      userData.idfa =
+        get(message, "context.device.advertisingId") ||
+        get(message, "context.idfa");
+      userData.idfv =
+        get(message, "context.device.id") || get(message, "context.idfv");
+    } else if (os.toLowerCase() === "android") {
+      userData.android_id =
+        get(message, "context.device.id") || get(message, "context.android_id");
+      userData.aaid =
+        get(message, "context.device.advertisingId") ||
+        get(message, "context.aaid");
```

Comment:
For `idfa`, `idfv` and `aaid` fields, let's keep the previous mapping as default and higher in the order.

Reply:
IMO since taking this config from the 1st precedence is the standard way for all other dest and in docs we have and the 2nd precedence is just for back compatibility. Keeping this way may boost the user to use the standard one instead of the deprecated ones?

Reply:
If we always prefer the new source locations, it wouldn't be exactly backward compatible. It makes sense when we are building something new. 
As an immediate effect, it might be a breaking change. Rather, we can take the softer route where the docs point only to the recommended ones.